code/modules/almayer/weaponhits.dm

```diff
@@ -26,36 +33,48 @@
 				if(salvo == TRUE)
 					var/shotspacing
 					for(var/turf/picked_atom in location)
-						addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), picked_atom, 400, 10, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data), shotspacing SECONDS)
+						addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), picked_atom, 700, 10, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data), shotspacing SECONDS)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_MISSILE_DAMAGE
 						shotspacing += 1
 						shakeship(10, 10, TRUE, FALSE)
+						GLOB.ship_hits_tally["times_hit_missile"] += 1
+						logtotalhits += 1
 					weaponhits_effects(WEAPON_MISSILE)
 				else
-					cell_explosion(location, 350, 1, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data)
+					cell_explosion(location, 700, 9, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data)
+					GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_MISSILE_DAMAGE
 					shakeship(10, 10, TRUE, FALSE)
 					weaponhits_effects(WEAPON_MISSILE)
+					GLOB.ship_hits_tally["times_hit_missile"] += 1
+					logtotalhits +=1
 			if(point_defense == TRUE)
 				var/hitchance = HIT_CHANCE_STANDARD
 				if(salvo == TRUE)
 					var/confirmedhit
 					var/shotspacing
 					for(var/turf/picked_atom in location)
 						if(prob(hitchance))
-							addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), picked_atom, 400, 10, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data), shotspacing SECONDS)
+							addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), picked_atom, 700, 9, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, ashm_cause_data), shotspacing SECONDS)
+							GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_MISSILE_DAMAGE
 							shakeship(10, 10, TRUE, FALSE)
 							confirmedhit += 1
+							logtotalhits += 1
 						else
 							weaponhits_effects(WEAPON_MISSILE, TRUE, shotspacing)
 
 						shotspacing += 1
 					if(confirmedhit > 0)
```

Comment:
If you don't intend to ever have multiple types of weapons hitting in one call, then you don't even need confirmedhit and instead could use logtotalhits.